web/frontend/src/components/utils/useConfiguration.tsx

```diff
@@ -0,0 +1,165 @@
+import { renderRank, renderSelect, renderSubject, renderText } from 'pages/ballot/QuestionDisplay';
+import { useEffect, useState } from 'react';
+import {
+  Answers,
+  Configuration,
+  Error,
+  ID,
+  Question,
+  RANK,
+  ROOT_ID,
+  RankAnswer,
+  SELECT,
+  SUBJECT,
+  SelectAnswer,
+  Subject,
+  TEXT,
+  TextAnswer,
+} from 'types/configuration';
+
+// Flattens and re-orders the questions (as the first object encountered in a
+// Subject might not be the first that needs to be displayed according to the Order).
+// Adds a ParentID for later rendering and initializes the Answers.
+function flattenSubject(
+  subject: Subject,
+  parentId: ID,
+  sortedQuestions: Array<Question>,
```

Comment:
this array is not sorted, but contains elements that have a sorted "Order" property. Am I correct ? If that's the case then the naming is misleading.